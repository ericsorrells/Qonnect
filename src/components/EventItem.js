// ========================================================================================
import React     from 'react';
import PropTypes from 'prop-types';
import moment    from 'moment';
import { Link }  from 'react-router-dom';
// ========================================================================================
import Acceptances from './EventItem_Acceptances';
// ========================================================================================
import { formatTime, formatDate, formatText } from '../utils/utils';
// ========================================================================================

const EventItem = (event) => {
  const { id, uid, userName, imageUrl, eventName, location, description, time, date, history, interestedUsers, deleteEvent } = event;

  const deleteElement = () => {
    deleteEvent(id);                            // passed from showEvents props
  }

  const editElement = () => {
    history.push(`/edit-event/${id}`);          // passed from showEvents props
  }

  const showElement = () => {
    history.push(`/show-event/${id}`);
  }

  return (
    <li className='event-item'>
      <div className='event-item__left-container'>
        <div className='event-item__left'>
          <Image Link={Link} id={id} imageUrl={imageUrl} />
        </div>
        <div className='event-item__center'>
          <Link to={`/show-event/${id}`}>
            <Info value={eventName} type={'title'} />
          </Link>
          <Link to={`/profile/${uid}`}>
            <Info value={userName} type={'inviter'} />
          </Link>
          <div className='event-item__date_location'>
            {location && <SubInfo value={formatText(location, 20)} value2={null} type={'location'} icon={'map-marker'} />}
            {date && time && <SubInfo value={formatDate(date)} value2={`| ${formatTime(time)}`} type={'date'} icon={'calendar'} />  }
          </div>
          <Info value={formatText(description, 250)} type={'description'} />
        </div>
      </div>
      <div className='event-item__acceptances'>
        {interestedUsers && <Acceptances acceptances={Object.keys(interestedUsers).length} />}
      </div>
    </li>
  )
}

const Image = ({Link, id, imageUrl}) => {
 return(
    <Link to={`/show-event/${id}`}>
      {
        imageUrl ?
          <img src={imageUrl} className='event-item__image' /> :
          <img src='/images/default.png' className='event-item__image' />
      }
    </Link>
 )
}

const Info = ({ value, type }) => {
  return (
    <div className={`event-item__${type}`}>
      {value}
    </div>
  )
}

const SubInfo = ({ value, value2, type, icon }) => {
  return (
    <div className={`event-item__${type}`}>
      <div style={{ fontSize: '1em', color: '#6334e3' }}>
        <i className={`fas fa-${icon}`}></i>
      </div>
      <div>{value}   {value2}</div>
    </div>
  )
}

EventItem.propTypes = {
  id:          PropTypes.number,
  url:         PropTypes.string,
  name:        PropTypes.string,
  location:    PropTypes.string,
  description: PropTypes.string,
  inviter:     PropTypes.string,
  date:        PropTypes.string,
  history:     PropTypes.func,
  deleteEvent: PropTypes.func
};

export default EventItem;