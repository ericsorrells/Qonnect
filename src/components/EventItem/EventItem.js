import React from 'react';

const EventItem = ({id, url, name, location, description, inviter, date, history}) => {
  const deleteElement = () => {
    console.log('DELETE ELEMENT CALLED!!!!', id);
  }

  const editElement = () => {
    history.push(`/edit-event/${id}`);          // passed from showEvents props
  }

  return (
    <li className="event-item">
      <div className="event-item__left">
        <img src={url} className="event-item__image" />
      </div>
      <div className="event-item__right">
        <div className="event-item__title">
          {name}
        </div>
        <div className="event-item__inviter">
          {inviter}
        </div>
        <div className="event-item__date_location">
          <div className="event-item__location">
            <div style={{ fontSize: '1em', color: '#6334e3' }}>
              <i className="fas fa-map-marker"></i>
            </div>
            <div>{location}</div>
            </div>
          <div className="event-item__date">
            <div style={{ fontSize: '1em', color: '#6334e3' }}>
              <i class="fas fa-calendar"></i>
            </div>
            <div>{date}</div>
          </div>
        </div>
        <div className="event-item__description">
          {description}
        </div>
      </div>
      <div className="hover-container">
        <div className="hover-buttons">
          <button className="edit-button" onClick={editElement}>Edit</button>
          <button className="delete-button" onClick={deleteElement}>Delete</button>
        </div>
      </div>
    </li>
  )
}

// const HoverButtons = () => {
//   return(
    
//   )
// };

export default EventItem;