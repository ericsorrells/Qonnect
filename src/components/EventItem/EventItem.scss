.event-item {
  width: 100%;
  border: 1px solid black;
  margin-bottom: 30px;
  display: flex;
  flex-direction: row;
  height: 150px;
  box-shadow: 10px 10px 19px 0px rgba(0,0,0,0.32);
  // overlay settings
  position: relative;                                  
  transition: .5s ease;
}

// overlay settings
.event-item:hover .event-item__left {
  opacity: 0.3;
}

// overlay settings
.event-item:hover .event-item__right {
  opacity: 0.3;
}

// overlay settings
.event-item:hover .hover-container {
  opacity: 1;
  display: inline-block;
}

// overlay settings
.hover-container{
  display: none;                       // gets toggled in component
  position: absolute;                  // positions absolutely w/in parent   
  top: 50%;                            // centers the buttons
  left: 50%;
  transform: translate(-50%, -50%);
  transition: .5s ease;
  opacity: 0;
}

// overlay settings
.hover-buttons {
  display: flex;
}

.edit-button,
.delete-button {
  margin-right: 50px;
  background-color: $purple;
  min-width: 100px;
  color: white;
  padding: 10px 30px;
  border: none;
  border-radius: 5px;
}

.event-item__left {
  display: flex;
  justify-content: center;
  align-items: center;
  overflow: hidden;
  width: 500px;
}

.event-item__left img {
  width: auto;
  height: 150px;
}

.event-item__right {
  padding: 10px;
}

.event-item__image {
  object-fit: cover;
}

.event-item__inviter {
  margin-top: 5px;
}

.event-item__title {
  font-weight: bold;
  text-transform: uppercase;
}

.event-item__date_location {
  color: $grey-text;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  border-bottom: 1px solid black;
  padding-bottom: 5px;
  margin-bottom: 5px;
  margin-top: 5px;
  font-size: .9rem;
}

.event-item__location,
.event-item__date {
  display: flex;
  flex-direction: row;
}

.event-item__description {
  color: $grey-text;
  font-size: .8rem;
}

.fa-calendar,
.fa-map-marker {
  margin-right: 10px;
}