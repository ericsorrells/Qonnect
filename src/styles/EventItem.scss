.event-item {
  width: 100%;
  border: 1px solid black;
  margin-bottom: 30px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  height: 150px;
  box-shadow: 10px 10px 19px 0px rgba(0,0,0,0.32);
}

.event-item__left-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 80%;
}

.event-item__left {
  justify-content: center;
  align-items: center;
  overflow: hidden;
  width: 200px;
}

.event-item__left img {
  width: auto;
  height: 150px;
}

.event-item__center {
  padding: 10px;
  width: 65%;
}

.event-item__acceptances {
  width: 20%;
}

.event-item__image {
  object-fit: cover;
}

.event-item__inviter {
  margin-top: 5px;
  color: black;
}

.event-item__title {
  font-weight: bold;
  text-transform: uppercase;
  color: black;
}

.event-item__date_location {
  color: $grey-text;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  border-bottom: 1px solid black;
  padding-bottom: 5px;
  margin-bottom: 5px;
  margin-top: 5px;
  font-size: .9rem;
}

.event-item__location,
.event-item__date {
  display: flex;
  flex-direction: row;
}

.event-item__description {
  color: $grey-text;
  font-size: .8rem;
}

.fa-calendar,
.fa-map-marker {
  margin-right: 10px;
}