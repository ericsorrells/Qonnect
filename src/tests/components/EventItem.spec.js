// ========================================================================================
import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import 'jest-enzyme';
// ========================================================================================
import EventItem, { Image } from '../../components/EventItem';
// ========================================================================================
import { basicEvent } from '../stubs/event';
import { basicProfile } from '../stubs/profile'
// ========================================================================================

describe('<EventItem />', () => {
  it('renders correctly with a valid event', () => {
    const component = shallow(
      <EventItem
        {...basicEvent}
        key={basicEvent.id}
        history={jest.fn()}
        deleteEvent={jest.fn()}
      />
    )

    expect(component).toMatchSnapshot();
  });

  it('renders default image when no imageURL provided', () => {
    const newEvent = { ...basicProfile, imageUrl: '' }
    const component = mount(
      <MemoryRouter>
        <Image Link={MemoryRouter} id={'abc123'} imageUrl={null} />
      </MemoryRouter>
    )

    expect(component.find('.event-item__image')).toHaveProp('src', '/images/default.png');
  });
});
