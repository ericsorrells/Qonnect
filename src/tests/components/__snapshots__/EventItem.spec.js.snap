// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventItem /> renders correctly with a valid event 1`] = `
<li
  className="event-item"
>
  <div
    className="event-item__left-container"
  >
    <div
      className="event-item__left"
    >
      <Image
        Link={[Function]}
        imageUrl="https://image.com"
      />
    </div>
    <div
      className="event-item__center"
    >
      <Link
        replace={false}
        to="/show-event/undefined"
      >
        <Info
          type="title"
          value="Cool Event Title"
        />
      </Link>
      <Link
        replace={false}
        to="/profile/abc123"
      >
        <Info
          type="inviter"
          value="Rachel Deckard"
        />
      </Link>
      <div
        className="event-item__date_location"
      >
        <SubInfo
          icon="map-marker"
          type="location"
          value="123 Elm St."
          value2={null}
        />
        <SubInfo
          icon="calendar"
          type="date"
          value="Sun Jan 4 '70"
          value2="| 02:13 AM"
        />
      </div>
      <Info
        type="description"
        value="Lorem ipsum text"
      />
    </div>
  </div>
  <div
    className="event-item__acceptances"
  >
    <EventItemAcceptances
      acceptances={4}
    />
  </div>
</li>
`;
